core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -119,7 +119,7 @@ private[spark] class SparkArithmeticException(
     errorClass: String,
     errorSubClass: Option[String] = None,
     messageParameters: Array[String],
-    queryContext: String = "")
+    queryContext: Option[QueryContext] = None)
```

Comment:
We should pass both `QueryContext` and `summary: String`

Reply:
I think it would be better to pass one thing. I created some kind of enriched QueryContext which can give us textual summary.